packages/framework/src/Model/SocialNetwork/SocialNetworkFacade.php

```diff
@@ -0,0 +1,97 @@
+
+namespace Shopsys\FrameworkBundle\Model\SocialNetwork;
+
+use App\Model\Customer\User\CustomerUserFacade;
+use App\Model\Customer\User\RegistrationDataFactory;
+use App\Model\Customer\User\RegistrationFacade;
+use App\Model\Security\LoginAsUserFacade;
+use Hybridauth\Exception\InvalidArgumentException;
+use Hybridauth\Exception\UnexpectedValueException;
+use Hybridauth\Hybridauth;
+use Hybridauth\User\Profile;
+use Shopsys\FrameworkBundle\Model\Administrator\Exception\DuplicateUserNameException;
+use Shopsys\FrameworkBundle\Model\SocialNetwork\Exception\SocialNetworkLoginException;
+use Symfony\Bridge\Monolog\Logger;
+use Symfony\Component\Validator\Constraints\Email;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class SocialNetworkFacade
+{
+    /**
+     * @param \App\Model\Customer\User\RegistrationDataFactory $registrationDataFactory
+     * @param \App\Model\Customer\User\RegistrationFacade $registrationFacade
+     * @param \Shopsys\FrameworkBundle\Model\SocialNetwork\SocialNetworkConfigFactory $socialNetworkConfigFactory
+     * @param \Symfony\Bridge\Monolog\Logger $logger
+     * @param \App\Model\Customer\User\CustomerUserFacade $customerUserFacade
+     * @param \App\Model\Security\LoginAsUserFacade $loginAsUserFacade
+     * @param \Symfony\Component\Validator\Validator\ValidatorInterface $validator
+     */
+    public function __construct(
+        protected readonly RegistrationDataFactory $registrationDataFactory,
+        protected readonly RegistrationFacade $registrationFacade,
+        protected readonly SocialNetworkConfigFactory $socialNetworkConfigFactory,
+        protected readonly Logger $logger,
+        protected readonly CustomerUserFacade $customerUserFacade,
+        protected readonly LoginAsUserFacade $loginAsUserFacade,
+        protected readonly ValidatorInterface $validator,
+    ) {
+    }
+
+    /**
+     * @param string $type
+     * @param string $redirectUrl
+     * @return array{accessToken: string, refreshToken: string}
+     */
+    public function login(string $type, string $redirectUrl): array
+    {
+        try {
+            $config = $this->socialNetworkConfigFactory->createConfig($redirectUrl);
+            $hybridauth = new Hybridauth($config);
+
+            $adapter = $hybridauth->authenticate($type);
+            $userProfile = $adapter->getUserProfile();
+
+            $this->validateDataFromSocialNetwork($userProfile);
+
+            $registrationData = $this->registrationDataFactory->createFromSocialNetworkProfile($userProfile);
+
+            try {
+                $customerUser = $this->registrationFacade->register($registrationData);
+            } catch (DuplicateUserNameException $exception) {
+                $customerUser = $this->customerUserFacade->findCustomerUserByEmailAndDomain($registrationData->email, $registrationData->domainId);
+            }
```

Comment:
let's discuss this logic, please. I was thinking about reverting the logic - first try to find the existing customer and if not found, register the customer. But I see the customer is being found inside the register method so I am not sure. Let's talk about it, please

Reply:
as we discussed, I left these changes as it is now. If you will change your mind, let me know 😊